Source/CarthageKit/Xcode.swift

```diff
@@ -587,11 +587,24 @@ public func buildScheme(_ scheme: String, withOptions options: BuildOptions, inP
 			}
 		}
 		.flatMapTaskEvents(.concat) { builtProductURL -> SignalProducer<URL, CarthageError> in
-			return createDebugInformation(builtProductURL)
-				.then(SignalProducer<URL, CarthageError>(value: builtProductURL))
-		}
+			return UUIDsForFramework(builtProductURL)
+				.collect()
+				.flatMap(.concat) { uuids -> SignalProducer<URL, CarthageError> in
+					// Only attempt to create debug info if there is at least 
+					// one dSYM architecture UUID in the framework. This can 
+					// occur if the framework is a static framework packaged 
+					// like a dynamic framework.
+					if uuids.isEmpty {
+						return SignalProducer(value: builtProductURL)
+					}
+
+					return createDebugInformation(builtProductURL)
+						.then(SignalProducer<URL, CarthageError>(value: builtProductURL))
+				}
```

Comment:
Would you mind altering this slightly to move the `.then` outside?

```swift
			return UUIDsForFramework(builtProductURL)
				.collect()
				.flatMap(.concat) { uuids -> SignalProducer<URL, CarthageError> in
					// Only attempt to create debug info if there is at least 
					// one dSYM architecture UUID in the framework. This can 
					// occur if the framework is a static framework packaged 
					// like a dynamic framework.
					if uuids.isEmpty {
						return .empty
					}

					return createDebugInformation(builtProductURL)
				}
				.then(SignalProducer<URL, CarthageError>(value: builtProductURL))
```

(The return type in that `flatMap` is most likely not right now.)